NullPointerException reading mongodb bson date (native image)

Hi,

only native image of my application cause  a NullPointerException parsing mongodb document with date field.

The error occours both with quarkus 1.6.0.final and 1.7.0.final. 

Here the full staktrace:
```
2020-08-21 15:12:12,711 ERROR [org.jbo.res.res.i18n] (executor-thread-1) RESTEASY002020: Unhandled asynchronous exception, sending back 500: org.jboss.resteasy.spi.ApplicationException: java.lang.NullPointerException
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:180)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:638)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:504)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:454)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:456)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:417)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:391)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:488)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:259)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:160)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:163)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:245)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:132)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:37)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:94)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.NullPointerException
	at java.text.SimpleDateFormat.matchZoneString(SimpleDateFormat.java:1695)
	at java.text.SimpleDateFormat.subParseZoneString(SimpleDateFormat.java:1763)
	at java.text.SimpleDateFormat.subParse(SimpleDateFormat.java:2169)
	at java.text.SimpleDateFormat.parse(SimpleDateFormat.java:1541)
	at org.bson.json.JsonReader.visitISODateTimeConstructor(JsonReader.java:836)
	at org.bson.json.JsonReader.readBsonType(JsonReader.java:225)
	at org.bson.codecs.BsonDocumentCodec.decode(BsonDocumentCodec.java:85)
	at org.bson.codecs.BsonDocumentCodec.decode(BsonDocumentCodec.java:42)
	at org.bson.codecs.BsonDocumentCodec.readValue(BsonDocumentCodec.java:104)
	at org.bson.codecs.BsonDocumentCodec.decode(BsonDocumentCodec.java:87)
	at org.bson.BsonDocument.parse(BsonDocument.java:63)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.executeUpdate(MongoOperations.java:619)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.update(MongoOperations.java:614)
	at it.cineca.model.elaborato.HeartBeat.update(HeartBeat.java)
	at it.cineca.manager.HeartBeatManager.aggiornaBattito(HeartBeatManager.java:64)
	at it.cineca.manager.HeartBeatManager_ClientProxy.aggiornaBattito(HeartBeatManager_ClientProxy.zig:216)
	at it.cineca.web.BeatController.aggiornaBattito(BeatController.java:38)
	at java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:167)
	... 29 more
```
I'm compiling native image using maven's quarkus standard configuration with these Resource Configuration File:

```
{
  "bundles": [  {"name":"sun.util.resources.TimeZoneNames"}   ],
  "resources": [     {       "pattern": "org/joda/time/tz/data/.*/.*$"     },
                            {       "pattern": "org/joda/time/tz/data/.*$"     }   ]
}
```
Thank you
Nicola

good morning,

I have prepared a simple test case [eventi-testcase.zip](https://github.com/quarkusio/quarkus/files/5116906/eventi-testcase.zip) with only four classes:
```
├── pom.xml
└── src
    └── main
        ├── java
        │   └── it
        │       └── cineca
        │           ├── BeatController.java
        │           └── HeartBeat.java
        └── resources
            ├── application.properties
            └── resources-config.json
```
Please note that the only dependencies in the pom file, besides quarkus, are jodatime and lombock.


The mongodb collection we try unsuccessfully to update contains documents like:
```json
// collection: heart_beats
{
	"_id" : "fffff124-615d-40b2-bb08-a7edd9afacd4",
	"interruzione" : false,
	"istante" : ISODate("2019-08-26T17:38:20.917+02:00")
}
```
This is my version of graalvm and my request command:

```
> java --version
openjdk 11.0.7 2020-04-14
OpenJDK Runtime Environment GraalVM CE 20.1.0 (build 11.0.7+10-jvmci-20.1-b02)
OpenJDK 64-Bit Server VM GraalVM CE 20.1.0 (build 11.0.7+10-jvmci-20.1-b02, mixed mode, sharing)
> curl --location --request PUT 'http://localhost:8080/v1/beat/ffff5e8a-2b59-4787-8386-fe039e8dd622?idModulo=1525e070-16a5-45c3-a8ed-72a649a03d18'
```
The stack trace is basically the same as in my previous post. I report it here for completeness:
```
2020-08-24 11:13:50,591 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /v1/beat/ffff5e8a-2b59-4787-8386-fe039e8dd622?idModulo=1525e070-16a5-45c3-a8ed-72a649a03d18 failed, error id: f17277fe-c49b-4dcb-928d-75a55f6cf2a0-1: org.jboss.resteasy.spi.UnhandledException: java.lang.NullPointerException
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:216)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:515)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:259)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:160)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:163)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:245)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:132)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:37)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:94)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.NullPointerException
	at java.text.SimpleDateFormat.matchZoneString(SimpleDateFormat.java:1695)
	at java.text.SimpleDateFormat.subParseZoneString(SimpleDateFormat.java:1763)
	at java.text.SimpleDateFormat.subParse(SimpleDateFormat.java:2169)
	at java.text.SimpleDateFormat.parse(SimpleDateFormat.java:1541)
	at org.bson.json.JsonReader.visitISODateTimeConstructor(JsonReader.java:836)
	at org.bson.json.JsonReader.readBsonType(JsonReader.java:225)
	at org.bson.codecs.BsonDocumentCodec.decode(BsonDocumentCodec.java:85)
	at org.bson.codecs.BsonDocumentCodec.decode(BsonDocumentCodec.java:42)
	at org.bson.codecs.BsonDocumentCodec.readValue(BsonDocumentCodec.java:104)
	at org.bson.codecs.BsonDocumentCodec.decode(BsonDocumentCodec.java:87)
	at org.bson.BsonDocument.parse(BsonDocument.java:63)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.executeUpdate(MongoOperations.java:619)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.update(MongoOperations.java:614)
	at it.cineca.HeartBeat.update(HeartBeat.java)
	at it.cineca.BeatController.aggiornaBattito(BeatController.java:21)
	at java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:167)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:638)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:504)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:454)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:456)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:417)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:391)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:488)
```
Thanks  you for your attention
Nicola

Hello,

please forgive the delay in answering but we are under an important deadline at work.

Unfortunately I was not even able to compile my test case with the version provided. This is the stack trace:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.7.0.Final:build (default) on project eventi: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.mongodb.panache.deployment.BasePanacheMongoResourceProcessor#registerJacksonSerDeser threw an exception: java.lang.InstantiationError
[ERROR] 	at java.base/jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48)
[ERROR] 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:906)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] 
[ERROR] 	[error]: Build step io.quarkus.mongodb.panache.deployment.BasePanacheMongoResourceProcessor#registerForReflection threw an exception: java.lang.InstantiationError
[ERROR] 	at java.base/jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48)
[ERROR] 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:906)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] 
[ERROR] 	[error]: Build step io.quarkus.mongodb.panache.deployment.BasePanacheMongoResourceProcessor#registerJsonbSerDeser threw an exception: java.lang.InstantiationError
[ERROR] 	at java.base/jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48)
[ERROR] 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:906)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] 
[ERROR] 	[error]: Build step io.quarkus.mongodb.panache.deployment.BasePanacheMongoResourceProcessor#unremovableClients threw an exception: java.lang.InstantiationError
[ERROR] 	at java.base/jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48)
[ERROR] 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:906)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] -> [Help 1]
```


I could have done something wrong, in fact I was able to compile and install locally the quarkus branch provided to me (999-SNAPSHOT) but only skipping the tests :-(

Good morning,

added the parameters indicated, the build was successful. We have a different error, still in the same SimpleDateFormat class.
```
2020-09-23 08:55:56,455 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /v1/beat/ffff5e8a-2b59-4787-8386-fe039e8dd622?idModulo=1525e070-16a5-45c3-a8ed-72a649a03d18 failed, error id: 6365b73f-66a9-4991-977c-4e111faa2961-1: org.jboss.resteasy.spi.UnhandledException: java.lang.NullPointerException
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:216)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:515)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:259)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:160)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:163)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:245)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:131)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:37)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:94)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.Good morning,
EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.NullPointerException
	at java.text.SimpleDateFormat.matchZoneString(SimpleDateFormat.java:1695)
	at java.text.SimpleDateFormat.subParseZoneString(SimpleDateFormat.java:1763)
	at java.text.SimpleDateFormat.subParse(SimpleDateFormat.java:2169)
	at java.text.SimpleDateFormat.parse(SimpleDateFormat.java:1541)
	at org.bson.json.JsonReader.visitISODateTimeConstructor(JsonReader.java:836)
	at org.bson.json.JsonReader.readBsonType(JsonReader.java:225)
	at org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:167)
	at org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:46)
	at org.bson.codecs.DocumentCodec.readValue(DocumentCodec.java:258)
	at org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:169)
	at org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:46)
	at org.bson.Document.parse(Document.java:110)
	at org.bson.Document.parse(Document.java:95)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.executeUpdate(MongoOperations.java:622)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.update(MongoOperations.java:617)
	at it.cineca.HeartBeat.update(HeartBeat.java)
	at it.cineca.BeatController.aggiornaBattito(BeatController.java:22)
	at java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:167)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:638)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:504)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:454)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:456)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:417)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:391)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:488)
	... 19 more
```

Please Note: I'm using the same test case I posted earlier this month, building with the command: 
```./mvnw package -Pnative -DskipTests -Dquarkus.platform.version=999-SNAPSHOT -Dquarkus-plugin.version=999-SNAPSHOT -Dquarkus.platform.artifact-id=quarkus-bom```

--- a/extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java
+++ b/extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java
@@ -1,6 +1,5 @@
 package io.quarkus.mongodb.panache.binder;
 
-import java.text.SimpleDateFormat;
 import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
@@ -9,7 +8,6 @@ import java.time.format.DateTimeFormatter;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Date;
-import java.util.TimeZone;
 import java.util.UUID;
 import java.util.stream.Collectors;
 
@@ -29,6 +27,8 @@ final class CommonQueryBinder {
     }
 
     static String escape(Object value) {
+        value = replaceWithInstant(value);
+
         if (value == null) {
             return "null";
         }
@@ -38,23 +38,9 @@ final class CommonQueryBinder {
         if (Number.class.isAssignableFrom(value.getClass()) || value instanceof Boolean) {
             return value.toString();
         }
-        if (value instanceof Date) {
-            SimpleDateFormat dateFormat = new SimpleDateFormat(ISO_DATE_PATTERN);
-            dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
-            Date dateValue = (Date) value;
-            return "ISODate('" + dateFormat.format(dateValue) + "')";
-        }
-        if (value instanceof LocalDate) {
-            LocalDate dateValue = (LocalDate) value;
-            return "ISODate('" + DateTimeFormatter.ISO_LOCAL_DATE.format(dateValue) + "')";
-        }
-        if (value instanceof LocalDateTime) {
-            LocalDateTime dateValue = (LocalDateTime) value;
-            return "ISODate('" + ISO_DATE_FORMATTER.format(dateValue.atZone(ZoneOffset.UTC)) + "')";
-        }
         if (value instanceof Instant) {
             Instant dateValue = (Instant) value;
-            return "ISODate('" + ISO_DATE_FORMATTER.format(dateValue.atZone(ZoneOffset.UTC)) + "')";
+            return "{\"$date\": \"" + ISO_DATE_FORMATTER.format(dateValue.atZone(ZoneOffset.UTC)) + "\"} ";
         }
         if (value instanceof UUID) {
             UUID uuidValue = (UUID) value;
@@ -65,6 +51,22 @@ final class CommonQueryBinder {
             return "ObjectId('" + objectId.toHexString() + "')";
         }
         return "'" + value.toString().replace("\\", "\\\\").replace("'", "\\'") + "'";
+    }
+
+    private static Object replaceWithInstant(Object value) {
+        if (value instanceof Date) {
+            Date dateValue = (Date) value;
+            value = dateValue.toInstant();
+        }
+        if (value instanceof LocalDate) {
+            LocalDate dateValue = (LocalDate) value;
+            value = dateValue.atStartOfDay(ZoneOffset.UTC).toInstant();
+        }
+        if (value instanceof LocalDateTime) {
+            LocalDateTime dateValue = (LocalDateTime) value;
+            value = dateValue.toInstant(ZoneOffset.UTC);
+        }
+        return value;
     }
 
     /**

--- a/extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java
+++ b/extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java
@@ -4,6 +4,7 @@ import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.Arrays;
 import java.util.Collection;
@@ -27,8 +28,6 @@ final class CommonQueryBinder {
     }
 
     static String escape(Object value) {
-        value = replaceWithInstant(value);
-
         if (value == null) {
             return "null";
         }
@@ -38,9 +37,21 @@ final class CommonQueryBinder {
         if (Number.class.isAssignableFrom(value.getClass()) || value instanceof Boolean) {
             return value.toString();
         }
+        if (value instanceof Date) {
+            ZonedDateTime zonedDateTime = ((Date) value).toInstant().atZone(ZoneOffset.UTC);
+            return "{\"$date\": \"" + ISO_DATE_FORMATTER.format(zonedDateTime) + "\"} ";
+        }
+        if (value instanceof LocalDate) {
+            ZonedDateTime zonedDateTime = ((LocalDate) value).atStartOfDay(ZoneOffset.UTC);
+            return "{\"$date\": \"" + ISO_DATE_FORMATTER.format(zonedDateTime) + "\"} ";
+        }
+        if (value instanceof LocalDateTime) {
+            ZonedDateTime zonedDateTime = ((LocalDateTime) value).atZone(ZoneOffset.UTC);
+            return "{\"$date\": \"" + ISO_DATE_FORMATTER.format(zonedDateTime) + "\"} ";
+        }
         if (value instanceof Instant) {
-            Instant dateValue = (Instant) value;
-            return "{\"$date\": \"" + ISO_DATE_FORMATTER.format(dateValue.atZone(ZoneOffset.UTC)) + "\"} ";
+            ZonedDateTime zonedDateTime = ((Instant) value).atZone(ZoneOffset.UTC);
+            return "{\"$date\": \"" + ISO_DATE_FORMATTER.format(zonedDateTime) + "\"} ";
         }
         if (value instanceof UUID) {
             UUID uuidValue = (UUID) value;
@@ -51,22 +62,6 @@ final class CommonQueryBinder {
             return "ObjectId('" + objectId.toHexString() + "')";
         }
         return "'" + value.toString().replace("\\", "\\\\").replace("'", "\\'") + "'";
-    }
-
-    private static Object replaceWithInstant(Object value) {
-        if (value instanceof Date) {
-            Date dateValue = (Date) value;
-            value = dateValue.toInstant();
-        }
-        if (value instanceof LocalDate) {
-            LocalDate dateValue = (LocalDate) value;
-            value = dateValue.atStartOfDay(ZoneOffset.UTC).toInstant();
-        }
-        if (value instanceof LocalDateTime) {
-            LocalDateTime dateValue = (LocalDateTime) value;
-            value = dateValue.toInstant(ZoneOffset.UTC);
-        }
-        return value;
     }
 
     /**

--- a/extensions/panache/mongodb-panache/runtime/src/test/java/io/quarkus/mongodb/panache/runtime/MongoOperationsTest.java
+++ b/extensions/panache/mongodb-panache/runtime/src/test/java/io/quarkus/mongodb/panache/runtime/MongoOperationsTest.java
@@ -50,18 +50,18 @@ class MongoOperationsTest {
         assertEquals("{'field':true}", query);
 
         query = operations.bindFilter(Object.class, "field", new Object[] { LocalDate.of(2019, 3, 4) });
-        assertEquals("{'field':ISODate('2019-03-04')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T00:00:00.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field", new Object[] { LocalDateTime.of(2019, 3, 4, 1, 1, 1) });
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field",
                 new Object[] { LocalDateTime.of(2019, 3, 4, 1, 1, 1).toInstant(ZoneOffset.UTC) });
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field",
                 new Object[] { toDate(LocalDateTime.of(2019, 3, 4, 1, 1, 1)) });
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field",
                 new Object[] { UUID.fromString("7f000101-7370-1f68-8173-70afa71b0000") });
@@ -90,19 +90,19 @@ class MongoOperationsTest {
 
         query = operations.bindFilter(Object.class, "{'field': ?1}",
                 new Object[] { LocalDate.of(2019, 3, 4) });
-        assertEquals("{'field': ISODate('2019-03-04')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T00:00:00.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': ?1}",
                 new Object[] { LocalDateTime.of(2019, 3, 4, 1, 1, 1) });
-        assertEquals("{'field': ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': ?1}",
                 new Object[] { LocalDateTime.of(2019, 3, 4, 1, 1, 1).toInstant(ZoneOffset.UTC) });
-        assertEquals("{'field': ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': ?1}",
                 new Object[] { toDate(LocalDateTime.of(2019, 3, 4, 1, 1, 1)) });
-        assertEquals("{'field': ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': ?1}",
                 new Object[] { UUID.fromString("7f000101-7370-1f68-8173-70afa71b0000") });
@@ -148,19 +148,19 @@ class MongoOperationsTest {
 
         query = operations.bindFilter(Object.class, "{'field': :field}",
                 Parameters.with("field", LocalDate.of(2019, 3, 4)).map());
-        assertEquals("{'field': ISODate('2019-03-04')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T00:00:00.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': :field}",
                 Parameters.with("field", LocalDateTime.of(2019, 3, 4, 1, 1, 1)).map());
-        assertEquals("{'field': ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': :field}",
                 Parameters.with("field", LocalDateTime.of(2019, 3, 4, 1, 1, 1).toInstant(ZoneOffset.UTC)).map());
-        assertEquals("{'field': ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': :field}",
                 Parameters.with("field", toDate(LocalDateTime.of(2019, 3, 4, 1, 1, 1))).map());
-        assertEquals("{'field': ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field': {\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "{'field': :field}",
                 Parameters.with("field", UUID.fromString("7f000101-7370-1f68-8173-70afa71b0000")).map());
@@ -209,18 +209,18 @@ class MongoOperationsTest {
         assertEquals("{'value':'a value'}", query);
 
         query = operations.bindFilter(Object.class, "field = ?1", new Object[] { LocalDate.of(2019, 3, 4) });
-        assertEquals("{'field':ISODate('2019-03-04')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T00:00:00.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = ?1", new Object[] { LocalDateTime.of(2019, 3, 4, 1, 1, 1) });
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = ?1",
                 new Object[] { LocalDateTime.of(2019, 3, 4, 1, 1, 1).toInstant(ZoneOffset.UTC) });
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = ?1",
                 new Object[] { toDate(LocalDateTime.of(2019, 3, 4, 1, 1, 1)) });
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = ?1",
                 new Object[] { UUID.fromString("7f000101-7370-1f68-8173-70afa71b0000") });
@@ -287,19 +287,19 @@ class MongoOperationsTest {
 
         query = operations.bindFilter(Object.class, "field = :field",
                 Parameters.with("field", LocalDate.of(2019, 3, 4)).map());
-        assertEquals("{'field':ISODate('2019-03-04')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T00:00:00.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = :field",
                 Parameters.with("field", LocalDateTime.of(2019, 3, 4, 1, 1, 1)).map());
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = :field",
                 Parameters.with("field", LocalDateTime.of(2019, 3, 4, 1, 1, 1).toInstant(ZoneOffset.UTC)).map());
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = :field",
                 Parameters.with("field", toDate(LocalDateTime.of(2019, 3, 4, 1, 1, 1))).map());
-        assertEquals("{'field':ISODate('2019-03-04T01:01:01.000Z')}", query);
+        assertEquals("{'field':{\"$date\": \"2019-03-04T01:01:01.000Z\"} }", query);
 
         query = operations.bindFilter(Object.class, "field = :field",
                 Parameters.with("field", UUID.fromString("7f000101-7370-1f68-8173-70afa71b0000")).map());

